### PR TITLE
refactor(node): making sn_node::Cmd non-clonable

### PR DESCRIPTION
**Pull request by bochaco**

### CODE REVIEW

File: sn_node/src/node/flow_ctrl/tests/cmd_utils.rs [hunk 1]
```diff
@@ -80,69 +79,87 @@ pub(crate) async fn handle_online_cmd(
...
     }
 
-    Ok(all_cmds)
-}
+    pub(crate) fn new_with_client_msg(
+        _msg: ClientMsg,
+        _peer: Peer,
+        dispatcher: &'a Dispatcher,
+    ) -> crate::node::error::Result<Self> {
+        // let node = dispatcher.node();
```
Comment (joshuef) [inline sn_node/src/node/flow_ctrl/tests/cmd_utils.rs:102]: bit of cleanup here

**Review by joshuef — APPROVED** [review]
bors r+

File: sn_node/src/node/flow_ctrl/tests/cmd_utils.rs [hunk 2]
```diff
@@ -80,69 +79,67 @@ pub(crate) async fn handle_online_cmd(
...
-pub(crate) async fn run_and_collect_cmds(
-    cmd: Cmd,
-    dispatcher: &Dispatcher,
-) -> crate::node::error::Result<Vec<Cmd>> {
-    let mut all_cmds = vec![];
-
-    let mut cmds = dispatcher.process_cmd(cmd).await?;
+pub(crate) struct ProcessAndInspectCmds<'a> {
+    pending_cmds: VecDeque<Cmd>,
+    cmds_to_inspect: VecDeque<usize>,
```
Comment (oetyng) [inline sn_node/src/node/flow_ctrl/tests/cmd_utils.rs:84]: Is this one really necessary?
Seems like we just push and pop from the collection. 
What do we use this additional storsge of indices for?

File: sn_node/src/node/flow_ctrl/tests/cmd_utils.rs [hunk 3]
```diff
@@ -80,69 +79,67 @@ pub(crate) async fn handle_online_cmd(
...
+    pub(crate) async fn next(&mut self) -> crate::node::error::Result<Option<&Cmd>> {
+        match self.cmds_to_inspect.pop_front() {
+            Some(index) => {
+                let cmd = self.pending_cmds.get(index);
+                assert!(cmd.is_some());
+                Ok(cmd)
+            }
+            None => {
+                while let Some(cmd) = self.pending_cmds.pop_front() {
+                    if !matches!(cmd, Cmd::SendMsg { .. }) {
```
Comment (oetyng) [inline sn_node/src/node/flow_ctrl/tests/cmd_utils.rs:118]: The fn and its signature can be simolified to Option<Cmd> by matching on the popped cmd, and skipping the index collection.

**Pull request merged**